datastore/writer/postgresql_backend/sql_database_backend_service.py

```diff
@@ -93,288 +140,133 @@ def create_position(
+        self, modified_collectionfields: Iterable[str], position: Position
     ):
         # insert into db, updating all existing fields with position, returning ids
-        value_placeholders = []
-        arguments: List[Any] = []
+        arguments = []
         for collectionfield in modified_collectionfields:
             if len(collectionfield) > COLLECTIONFIELD_MAX_LEN:
                 raise InvalidFormat(
-                    f"Collection field {collectionfield} is too long "
-                    + "(max: {COLLECTIONFIELD_MAX_LEN})"
+                    f"Collection field {collectionfield} is too long (max: {COLLECTIONFIELD_MAX_LEN})"
                 )
-            arguments.extend(
+            arguments.append(
                 (
                     collectionfield,
                     position,
                 )
             )
-            value_placeholders.append("(%s, %s)")
 
-        values = ",".join(value_placeholders)
         statement = dedent(
-            f"""\
-            insert into collectionfields (collectionfield, position) values {values}
+            """\
+            insert into collectionfields (collectionfield, position) values %s
             on conflict(collectionfield) do update set position=excluded.position
             returning id"""
         )
-        return self.connection.query_list_of_single_values(statement, arguments)
+        return self.connection.query_list_of_single_values(
+            statement, arguments, use_execute_values=True
+        )
 
-    def connect_events_and_collection_fields(self, event_id, collectionfield_ids):
-        # create m2m relations
-        value_placeholders = []
-        arguments: List[Any] = []
-        for collectionfield_id in collectionfield_ids:
+    def connect_events_and_collection_fields(
+        self,
+        event_ids: List[int],
+        collectionfield_ids: List[int],
+        event_indices_order: Iterable[List[int]],
+    ):
+        arguments: List[Tuple[int, int]] = []
+        for collectionfield_id, event_indices in zip(
+            collectionfield_ids, event_indices_order
+        ):
             arguments.extend(
                 (
-                    event_id,
+                    event_ids[event_index],
                     collectionfield_id,
                 )
+                for event_index in event_indices
             )
-            value_placeholders.append("(%s, %s)")
 
-        values = ",".join(value_placeholders)
-        statement = f"insert into events_to_collectionfields (event_id, collectionfield_id) values {values}"
-        self.connection.execute(statement, arguments)
+        statement = "insert into events_to_collectionfields (event_id, collectionfield_id) values %s"
+        self.connection.execute(statement, arguments, use_execute_values=True)
 
     def json(self, data):
-        return self.connection.to_json(data)
+        return self.connection.to_json(deepcopy(data))
```

Comment:
That previously was the case, see https://github.com/OpenSlides/openslides-datastore-service/blob/bfc77a4ba9d419a40a55ad36e252ec637cb28cdc/datastore/writer/postgresql_backend/sql_database_backend_service.py#L208 where `event.field_data` is passed as the data argument and then in https://github.com/OpenSlides/openslides-datastore-service/blob/bfc77a4ba9d419a40a55ad36e252ec637cb28cdc/datastore/writer/postgresql_backend/sql_database_backend_service.py#L141 the `META_DELETED` flag is set into the data. It was however intended since the changed data was used to calculate the modified fields which are sent over redis. but I changed the behaviour in this PR and therefore the deepcopy can be remove again, thanks for spotting this.